File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        if(n < 10) return false;
        int sum = 0, prod = 1, temp = n;
        while(temp > 0) {
            int rem = temp % 10;
            temp /= 10;
            sum += rem;
            prod *= rem;
        }
        return n % (sum+prod) == 0;
    }
}

File: 3918-check-divisibility-by-digit-sum-and-product/check-divisibility-by-digit-sum-and-product.go
func checkDivisibility(n int) bool {
    sum := 0; prod := 1; temp := n;
    for temp > 0 {
        rem := temp % 10
        temp /= 10
        sum += rem
        prod *= rem
    }

    return n % (sum + prod) == 0;
}